src/main/scala/cromwell/binding/WdlExpression.scala

```diff
@@ -84,7 +85,8 @@ object WdlExpression {
   }
 }
 
-case class WdlExpression(ast: AstNode) {
+case class WdlExpression(ast: AstNode) extends WdlValue {
+  val wdlType = WdlExpressionType
```

Comment:
If I'm looking at things correctly it ends up in some strings and there's an equality, correct?

The latter is easy, there's the type equality operator. Not sure about the toString though.